python/lsst/ts/externalscripts/make_LOVE_stress_tests.py

```diff
@@ -0,0 +1,299 @@
+
+import asyncio
+import json
+
+import aiohttp
+import requests
+import yaml
+from astropy.time import Time
+from lsst.ts import salobj
+
+
+class ClientInstance:
+    """Connect to a LOVE-manager instance.
+
+    Notes
+    -----
+    **Details**
+
+    * Generate websocket connections using provided credentials
+    by token authentication and subscribes to every
+    event and telemetry specified.
+    """
+
+    def __init__(self, location, username, password, event_streams, telemetry_streams):
+        self.location = location
+        self.username = username
+        self.password = password
+        self.event_streams = event_streams
+        self.telemetry_streams = telemetry_streams
+
+        self.websocket_url = ""
+        self.received_messages = 0
+        self.msg_traces = []
+
+    def request_token(self):
+        """Authenticate on the LOVE-manager instance
+        to get an authorization token."""
+
+        url = f"http://{self.location}/manager/api/get-token/"
+        data = {
+            "username": self.username,
+            "password": self.password,
+        }
+        resp = requests.post(url, data=data)
+        try:
+            token = resp.json()["token"]
+        except Exception:
+            raise Exception("Authentication failed")
+        self.websocket_url = (
+            f"ws://{self.location}/manager/ws/subscription?token={token}"
+        )
+
+    async def handle_message_reception(self):
+        """Handles the reception of messages."""
+
+        if self.websocket:
+            async for message in self.websocket:
+                if message.type == aiohttp.WSMsgType.TEXT:
+                    msg = json.loads(message.data)
+                    if "category" not in msg or (
+                        "option" in msg and msg["option"] == "subscribe"
+                    ):
+                        continue
+                    self.received_messages = self.received_messages + 1
+                    tracing = msg["tracing"]
+                    tracing["client_rcv"] = Time.now().tai.datetime.timestamp()
+                    self.msg_traces.append(tracing)
+
+    async def subscribe_to(self, csc, salindex, stream, topic_type):
+        """Subscribes to the specified stream"""
+
+        subscribe_msg = {
+            "option": "subscribe",
+            "category": topic_type,
+            "csc": csc,
+            "salindex": salindex,
+            "stream": stream,
+        }
+        await self.websocket.send_str(json.dumps(subscribe_msg))
+
+    async def start_ws_client(self):
+        """Start client websocket connection"""
+
+        async with aiohttp.ClientSession() as session:
+            self.websocket = await session.ws_connect(self.websocket_url)
+            for csc in self.event_streams:
+                for stream in self.event_streams[csc]:
+                    await self.subscribe_to(
+                        csc.split(":")[0], csc.split(":")[1], stream, "event"
```

Comment:
I gently suggest using `csc_name, sal_index = salobj.name_to_name_index(csc)` and calling this once right after `for csc in self.event_streams`. (or `salindex`; I tend to use snake_case in my code).